src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import classes from './Pagination.module.css';
+import { ReactComponent as NavigateNextIcon } from './navigate_next.svg';
+import { ReactComponent as NavigateBeforeIcon } from './navigate_before.svg';
+import { ReactComponent as FirstPageIcon } from './first_page.svg';
+import { ReactComponent as LastPageIcon } from './last_page.svg';
+
+export interface PaginationProps {
+  numberOfRows: number;
+  rowsPerPageOptions: number[];
+  rowsPerPage: number;
+  onRowsPerPageChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  currentPage: number;
+  setCurrentPage: React.Dispatch<React.SetStateAction<number>>;
+  rowsPerPageText: string;
+  pageDescriptionText: string;
+}
+
+export const Pagination = ({
+  numberOfRows,
+  rowsPerPageOptions,
+  rowsPerPage,
+  onRowsPerPageChange,
+  currentPage,
+  setCurrentPage,
+  rowsPerPageText,
+  pageDescriptionText,
+}: PaginationProps) => {
+  const [numberOfPages, setNumberOfPages] = useState(1);
+
+  useEffect(() => {
+    if (numberOfRows < rowsPerPage) setNumberOfPages(1);
+    else setNumberOfPages(Math.ceil(numberOfRows / rowsPerPage));
+  }, [rowsPerPage, numberOfRows]);
```

Comment:
For better clarity, I think statements inside if-else blocks should be wrapped in curly-braces.
```suggestion
  useEffect(() => {
    if (numberOfRows < rowsPerPage) {
      setNumberOfPages(1);
    } else {
      setNumberOfPages(Math.ceil(numberOfRows / rowsPerPage));
    }
  }, [rowsPerPage, numberOfRows]);
```

Reply:
I agree - but we should probably add `eslint` rules for this, to avoid manually enforcing a code style.